content/docs/command-reference/remote/modify.md

```diff
@@ -271,6 +271,45 @@ including obtaining the necessary credentials, and how to form `gdrive://` URLs.
   $ dvc remote modify myremote gdrive_client_secret <client secret>
   ```
 
+- `gdrive_use_service_account` - instructs DVC to authenticate using a
+  [service account](/doc/user-guide/setup-google-drive-remote#configure-the-google-drive-remote-with-a-service-account).
+
+  ```dvc
+  $ dvc remote modify myremote gdrive_use_service_account true
+  ```
+
+- `gdrive_service_account_email` - email address of the Google Project's service
+  account.
+
+  ```dvc
+  $ dvc remote modify myremote gdrive_service_account_email <service acct email>
+  ```
+
+- `gdrive_service_account_p12_file_path` - Google Project's service account
+  `.p12` file path.
+
+  ```dvc
+  $ dvc remote modify myremote gdrive_service_account_p12_file_path path/to/file.p12
+  ```
+
+- `gdrive_service_account_user_email` - email of the service account to
+  impersonate.
```

Comment:
Hmmm actually I still don't understand the difference between `gdrive_service_account_email` and `gdrive_service_account_user_email`. Will read about this [impersonation](https://medium.com/google-cloud/impersonating-users-with-google-cloud-platform-service-accounts-ba762db09092) thing...

Reply:
@maxhora I assume this is just an optional gdrive remote config param?

Reply:
the article reads "enables another user or service account to impersonate a service account". So who is impersonating who? `gdrive_service_account_email` impersonates `gdrive_service_account_user_email`? Also could the param names be less similar and more explicit e.g. change the latter for `gdrive_impersonated_service_account_email`?

Reply:
@jorgeorpinel from the referenced article this option allows to do following:

"But what if you want to do the inverse, ie. impersonate a user with a service account? This is a common use case in a service-based architecture: services will perform actions (eg. store a record, retrieve a blob) on behalf of users."

It might be also possible to impersonate service account with user account, but I haven't seen such option before at all. I would suggest just reference official Google Docs about impersonating. Mainly, `gdrive_service_account_user_email` was included into DVC only because PyDrive2 supports such feature and such feature might be usable to advanced GDrive users who already know why they need users impersonating.

Reply:
Got it, updated now.

But in the Google docs I see no refernces of service accounts impersonating users, only the other way around. Seems like a bit of a hack? E.g. see https://cloud.google.com/iam/docs/understanding-service-accounts

Reply:
p.s. I guess this one doc kind of hints at this feature: https://developers.google.com/admin-sdk/directory/v1/guides/delegation